core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/internal/UserStoreMgtDataHolder.java

```diff
@@ -47,4 +52,40 @@ public CryptoService getCryptoService() {
         return cryptoService;
     }
 
+    /**
+     * Set each HashProviderFactory instance in a Map.
+     *
+     * @param hashProviderFactory Instance of HashProviderFactory.
+     */
+    public void setHashProviderFactory(HashProviderFactory hashProviderFactory) {
+
+        if (hashProviderFactoryMap == null) {
+            hashProviderFactoryMap = new HashMap<>();
+        }
+        hashProviderFactoryMap.put(hashProviderFactory.getType(), hashProviderFactory);
+    }
+
+    /**
+     * Get the HashProviderFactory instance from Map.
+     *
+     * @param algorithm Algorithm name for respective instance of HashProviderFactory.
+     * @return The HashProviderFactory instance, null if there were no such instance from the algorithm name.
```

Comment:
`The HashProviderFactory instance which has the given algorithm as the type. The method will return NULL if there were no matching HashProviderFactory to the given algorithm.`

Reply:
Resolved.